File: Task-3/Javascript/SubTask-4.js
const fs = require('fs');

fs.readFile('input.txt', 'utf8', (err, data) => {
    if (err) throw err;

    const n = parseInt(data.trim(), 10);
    let diamond = '';

    if (Number.isInteger(n) && n > 0) {
        for (let i = 0; i < n; i++) {
            diamond += ' '.repeat(n - i - 1) + '*'.repeat(2 * i + 1) + '\n';
        }
        for (let i = n - 2; i >= 0; i--) {
            diamond += ' '.repeat(n - i - 1) + '*'.repeat(2 * i + 1) + '\n';
        }
    } else {
        diamond = 'Please enter a positive integer.';
    }

    fs.writeFile('output.txt', diamond, (err) => {
        if (err) throw err;
        console.log('Diamond pattern written to output.txt');
    });
});

File: Task-3/python/SubTask-4.py
with open('input.txt', 'r') as infile:
    x = int(infile.read().strip())

y = []

if x > 0:
    for i in range(x):
        line = ' ' * (x - i - 1) + '*' * (2 * i + 1)
        y.append(line)
    
    for i in range(x - 2, -1, -1):
        line = ' ' * (x - i - 1) + '*' * (2 * i + 1)
        y.append(line)
else:
    y.append("Please enter a positive integer.")

with open('output.txt', 'w') as outfile:
    for line in y:
        outfile.write(line + '\n')
